OPHD/UI/RobotInspector.cpp

```diff
@@ -0,0 +1,91 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include "RobotInspector.h"
+#include "TextRender.h"
+
+#include "../Cache.h"
+#include "../Constants.h"
+
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+
+
+using namespace NAS2D;
+
+
+namespace
+{
+	const NAS2D::Image& robotImage(RobotType robotType)
+	{
+		static const std::map<RobotType, const Image*> robotImages
+		{
+			{RobotType::ROBOT_DIGGER, &imageCache.load("ui/interface/product_robodigger.png")},
+			{RobotType::ROBOT_DOZER, &imageCache.load("ui/interface/product_robodozer.png")},
+			{RobotType::ROBOT_MINER, &imageCache.load("ui/interface/product_robominer.png")}
+		};
+		return *robotImages.at(robotType);
+	}
+}
+
+
+RobotInspector::RobotInspector() :
+	mFont{ fontCache.load(constants::FONT_PRIMARY, constants::FONT_PRIMARY_NORMAL) },
+	mFontBold{ fontCache.load(constants::FONT_PRIMARY_BOLD, constants::FONT_PRIMARY_NORMAL) },
+	btnCancelOrders{ "Cancel Orders" },
+	btnSelfDestruct{ "Self Destruct" },
+	btnIdle{ "Set Idle" },
+	btnCancel{ constants::ButtonCancel }
+{
+	init();
+}
```

Comment:
I am of the philosophy that the constructor should never have any code and if it does it should be relegated to a dedicated function that's called from the c'tor.

Reply:
Yes, but why?

Reply:
If I end up needing to create any overloaded c'tors, I can invoke the function instead of copy/pasting. That's the biggest reason.

Reply:
Yes, that's why I gave a link to **delegating constructors**. That's a problem that was dealt with in C++11.

Reply:
I managed to miss that link.

I guess you could say I'm still getting used to the new features of C++ since C++11. You'd think that a decade of work would give me a bit of a clue -_-

Reply:
C++11 was a bit update :wink: